packages/foundation-models/src/azure-openai/azure-openai-chat-completion-stream-chunk-response.ts

```diff
@@ -0,0 +1,39 @@
+import type { AzureOpenAiCompletionUsage } from './client/inference/schema/index.js';
+
+/**
+ * Azure OpenAI chat completion stream chunk response.
+ */
+export class AzureOpenAiChatCompletionStreamChunkResponse {
+  constructor(public readonly data: any) {
+    // TODO: Change `any` to `CreateChatCompletionStreamResponse` once the preview spec becomes stable.
+    this.data = data;
+  }
+
+  /**
+   * Usage of tokens in the chunk response.
+   * @returns Token usage.
+   */
+  getTokenUsage(): AzureOpenAiCompletionUsage {
+    return this.data.usage;
+  }
+
+  /**
+   * Reason for stopping the completion stream chunk.
+   * @param choiceIndex - The index of the choice to parse.
+   * @returns The finish reason.
+   */
+  getFinishReason(choiceIndex = 0): string | undefined {
+    return this.data.choices.find((c: any) => c.index === choiceIndex)
+      ?.finish_reason;
+  }
+
+  /**
+   * Parses the chunk response and returns the delta content.
+   * @param choiceIndex - The index of the choice to parse.
+   * @returns The message delta content.
+   */
+  getDeltaContent(choiceIndex = 0): string | undefined | null {
```

Comment:
It can be `null` according to the spec -> https://github.com/Azure/azure-rest-api-specs/blob/60c67f112c66537f1d006bf1b497857ccd2afacd/specification/cognitiveservices/data-plane/AzureOpenAI/inference/stable/2024-10-21/inference.yaml#L1985C11-L1985C25